treeherder/model/models.py

```diff
@@ -115,6 +115,48 @@ def __str__(self):
             self.name, self.repository_group)
 
 
+class Push(models.Model):
+    '''
+    A push to a repository
+
+    A push should contain one or more commit objects, representing
+    the changesets that were part of the push
+    '''
+    repository = models.ForeignKey(Repository, db_index=False)
+    revision_hash = models.CharField(max_length=50, null=True, unique=True)  # legacy
+    # revision can be null if revision_hash defined ^^
+    revision = models.CharField(max_length=40,
```

Comment:
I believe the case I wanted to support here was where we only had the revision hash, no revision. But maybe we don't need to support that?